testplan/testing/multitest/base.py

```diff
@@ -709,6 +709,8 @@ def stop_test_resources(self):
         interactive mode where a test environment may be stopped on-demand.
         Handles running before/after_stop callables if required.
         """
+        if self.report.env_status == entity.ResourceStatus.STOPPED:
```

Comment:
it seems report.env_status is not something that we handle at Test or Multitest level. It is currently handled by TestRunnerIHandler. Or maybe we can move this state transition logic to Test. Anyway, it's better we set and check in the same class.

Reply:
For now I addressed the other two comments. I think the Environment itself should know about the overall status of its drivers and derive from that. It does not seem to me a Test or MultiTest level thing, only if it derives from the actual environment.